### PR TITLE
Collection of changes (refactoring, enhancements, bug fixes)

### PR DESCRIPTION
**Pull request by maks-io**
As already mentioned in another issue or pull request, I was working on some enhancements/changes over the last days.

First of all, I gotta say that I am sorry that this is such a huge PR, containing so many different things. The reason for this is, that I mainly made all those changes for a team of mine, which needed to migrate a few repos from GitLab und Github, but I still think that at least most of those changes make sense to be added in this repo too.

I will try to give a brief overview now:

- Refactoring the entire code
I did this to get a better overview, to shorten the `index.js` file and to be able to quicker detect bugs etc. Now there is not only a `index.js` file, but also a `githubHelper.js` (this contains everything regarding Github), a `gitlabHelper.js` (this contains everything regarding GitLab) and a `utils.js` (which contains things that somehow belong to both platforms, or none). The `index.js` still contains the basic code and is still the entry point.
As part of this refactoring step I also switched to `es6` (see http://es6-features.org/ for example), and therefore the starting command changed to `npm run start` (as described in `README.md`).

- Placeholder issue toggle
Placeholder issues can now be toggled via the flag `usePlaceholderIssuesForMissingIssues` in `settings.js`, since we had a case where we didn't want placeholder issues to be created.

- Replacement issues
There are still bugs occurring when migrating some issues. The new logic will now add so called "replacement issues" to keep things in order in those cases (you can also turn this off by setting `useReplacementIssuesForCreationFails` in `settings.js` to `false`). They are pretty similar to placeholder issues, however, they are containing every info that the original issue contained, except for its description (because in our cases there was some `octokit` error occurring, somehow connected to the issue description).

- Issue migration statistics
At the end of the process a small overview is being printed to the console, showing how many issues were migrated successfully, how many placeholder issues were used, how many replacement issues were needed and how many issues failed.

- Custom string ignore
I added a prop `skipMatchingComments` in `settings.js`, which is explained in detail in `README.md`.

- null bug
At two points in the code a check occurred which was testing some variables if they were `null`. This led to errors, since the variables weren't `null` in some cases, but `undefined` instead.

- Other small changes
The code contained a lot of `==`- and `!=` comparison logic, which can lead to bugs and errors (keyword `Code Coercion`). I fixed that. Also I replaced some `var`s by `const` or `let`.

Please also test this with migrating repositories that contain pull requests/merge requests, since I had no access to any GitLab repo with merge requests.

### CODE REVIEW

**Comment by maks-io** [comment]
> The code looks good to me so far. I'd like to run it against some test repos before merging it.

Hi spruce, yes that's how I imagined it - please, if somehow possible, do some detailed testing before merging it. Thanks for that!
Also it would be worth thinking about adding automated testing in the near future.

File: gitlabHelper.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          projects[i].name,
+          '\t--\t',
+          projects[i].description
+        );
+      }
+
+      // instructions for user
+      console.log('\n\n');
+      console.log(
+        'Select which project ID should be transported to github. Edit the settings.js accordingly. (gitlab.projectID)'
```
Comment (spruce) [inline gitlabHelper.js]: The settings path is wrong. It should be `gitlab.projectId`. I just ran into it

**Comment by spruce** [comment]
It did migrate a confidential issue (no 3). Do you think there should be something stopping such a migration?

File: gitlabHelper.js (hunk 1 above)
Comment (maks-io) [reply gitlabHelper.js]: Nice find! Do you want me to correct it in this PR (since you mentioned it here)?

**Comment by maks-io** [comment]
> It did migrate a confidential issue (no 3). Do you think there should be something stopping such a migration?

To be honest, I didn't even know they existed in GitLab (I just googled it). I think it's worth thinking about it (maybe add some new logic and another flag in `settings.js` to let the user decide how to deal with them), but I would open a separate issue for that. This PR is huge already :D

File: gitlabHelper.js (hunk 1 above)
Comment (spruce) [reply gitlabHelper.js]: Yes, that'd be great.

**Comment by spruce** [comment]
> but I would open a separate issue for that. This PR is huge already :D
I agree.

**Comment by Cairo2k18** [comment]
I keep receiving this error whilst trying to run this..?

```
C:\Users\Administrator\Desktop\n-gl2gh\index.js:1
import GithubHelper from './githubHelper';
^^^^^^

SyntaxError: Cannot use import statement outside a module
[90m    at Module._compile (internal/modules/cjs/loader.js:881:18)[39m
[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:962:10)[39m
[90m    at Module.load (internal/modules/cjs/loader.js:798:32)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:711:12)[39m
[90m    at Function.Module.runMain (internal/modules/cjs/loader.js:1014:10)[39m
[90m    at internal/main/run_main_module.js:17:11[39m
```

**Comment by spruce** [comment]
@Cairo2k18 this seems to be the case on older node versions. Which version are you running? Could you create a new issue for that?

**Pull request merged**